contracts/BridgeCommittee.sol

```diff
@@ -52,13 +57,16 @@ contract BridgeCommittee {
         // Prepare the message hash
         bytes32 messageHash = getMessageHash(message);
         // Check that the message has not already been approved
-        require(messageApproved[messageHash], "BridgeCommittee: Message already approved");
+        // nit: `approved` => `processed`
+        require(!messageProcessed[messageHash], "BridgeCommittee: Message already approved");
 
+        // this does not match the encoding rules: `SUI_NATIVE_BRIDGE` should be prefixed to raw message instead of hash.
```

Comment:
Will need to explore how this should work using real messages but more than likely you're correct.